core/src/main/java/edu/wpi/grip/core/operations/Operations.java

```diff
@@ -1,47 +1,78 @@
 package edu.wpi.grip.core.operations;
 
+import com.google.common.collect.ImmutableList;
 import com.google.common.eventbus.EventBus;
+import com.google.inject.Inject;
+import com.google.inject.Singleton;
+import edu.wpi.grip.core.Operation;
 import edu.wpi.grip.core.events.OperationAddedEvent;
 import edu.wpi.grip.core.operations.composite.*;
-import edu.wpi.grip.core.operations.networktables.NTNumber;
-import edu.wpi.grip.core.operations.networktables.NTPublishOperation;
-import edu.wpi.grip.core.operations.networktables.NTVector2D;
+import edu.wpi.grip.core.operations.network.NumberPublishable;
+import edu.wpi.grip.core.operations.network.Vector2D;
+import edu.wpi.grip.core.operations.network.networktables.NTManager;
+import edu.wpi.grip.core.operations.network.networktables.NTPublishOperation;
+import edu.wpi.grip.core.operations.network.ros.ROSManager;
+import edu.wpi.grip.core.operations.network.ros.ROSPublishOperation;
 import edu.wpi.grip.core.operations.opencv.MatFieldAccessor;
 import edu.wpi.grip.core.operations.opencv.MinMaxLoc;
 import edu.wpi.grip.core.operations.opencv.NewPointOperation;
 import edu.wpi.grip.core.operations.opencv.NewSizeOperation;
 
-import static org.bytedeco.javacpp.opencv_core.*;
+import java.util.function.Supplier;
 
-public final class Operations {
+import static com.google.common.base.Preconditions.checkNotNull;
+import static org.bytedeco.javacpp.opencv_core.Point;
+import static org.bytedeco.javacpp.opencv_core.Size;
 
-    private Operations() { /* no op */}
+@Singleton
+public class Operations {
+    private final EventBus eventBus;
+    private final ImmutableList<Supplier<Operation>> operations;
 
-    public static void addOperations(EventBus eventBus) {
+    @Inject
+    Operations(EventBus eventBus, NTManager ntManager, ROSManager rosManager) {
+        this.eventBus = checkNotNull(eventBus, "EventBus cannot be null");
+        checkNotNull(ntManager, "ntManager cannot be null");
+        checkNotNull(rosManager, "rosManager cannot be null");
+        this.operations = ImmutableList.of(
```

Comment:
Because every time you call addOperations you need to create new instances of the class.
That's the design we've been using all along.
If you look at the old version thats what we have been doing already I assumed that we always wanted to create new instances each time `addOperation` was called.

And, I'm not "showing off" I'm taking advantage of the language features. This is perfectly clear java code.